app/services/mhv_accounts_service.rb

```diff
@@ -42,60 +41,49 @@ def create
   end
 
   def upgrade
-    if mhv_account.may_upgrade?
+    if mhv_account.upgradable?
       client_response = mhv_ac_client.post_upgrade(params_for_upgrade)
       if client_response[:status] == 'success'
         StatsD.increment("#{STATSD_ACCOUNT_UPGRADE_KEY}.success")
         mhv_account.upgraded_at = Time.current
+        Common::Collection.bust("#{mhv_account.mhv_correlation_id}:geteligibledataclass")
         mhv_account.upgrade!
       end
-    end
-  rescue => e
-    if e.is_a?(Common::Exceptions::BackendServiceException) && e.original_body['code'] == 155
+    elsif user.mhv_account.already_premium?
```

Comment:
Why do you have to precede it with `user` if you already have access to `mhv_account`?

Reply:
good point. ;)